### PR TITLE
Allow passing of directives to making scalars

### PR DESCRIPTION
**Pull request by yarian**
We ran into an issue when upgrading federation 2. We need to be able to add @tag directives to scalar type definitions. This currently doesn't seem supported but it seems like an oversight given that it's there for enums, etc...

### CODE REVIEW

File: core/src/main/scala/caliban/schema/Schema.scala [hunk 1]
```diff
@@ -137,11 +137,12 @@ trait GenericSchema[R] extends SchemaDerivation[R] with TemporalSchema {
     name: String,
     description: Option[String],
     specifiedBy: Option[String],
+    directives: Option[List[Directive]],
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/schema/Schema.scala:141]: Can you update the scaladoc?

File: core/src/main/scala/caliban/schema/Schema.scala [hunk 2]
```diff
@@ -131,17 +131,19 @@ trait GenericSchema[R] extends SchemaDerivation[R] with TemporalSchema {
    * @param name name of the scalar type
    * @param description description of the scalar type
    * @param specifiedBy URL of the scalar specification
+   * @param directives The directives to add to the type
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/schema/Schema.scala]:
```suggestion
   * @param directives the directives to add to the type
```
nitpicking just to be consistent with the other parameters 😄

File: core/src/main/scala/caliban/Rendering.scala [hunk 3]
```diff
@@ -20,7 +20,9 @@ object Rendering {
               if (isBuiltinScalar(name)) None
               else
                 Some(
-                  s"""${renderDescription(t.description)}scalar $name${renderSpecifiedBy(t.specifiedBy)}""".stripMargin
+                  s"""${renderDescription(t.description)}\n${renderDirectives(
```
Comment (ghostdogpr) [inline core/src/main/scala/caliban/Rendering.scala]: According to the spec, the directives come after the name: https://spec.graphql.org/October2021/#sec-Scalars

Comment (yarian) [reply core/src/main/scala/caliban/Rendering.scala]: Ah, TIL. Thanks will fix.

Comment (ghostdogpr) [reply core/src/main/scala/caliban/Rendering.scala]: Also I think you don't need any space or line break, if you look at other usage of `renderDirectives`

**Comment by yarian** [comment]
Looks like CircleCI got its act together 🥳

**Review by ghostdogpr — APPROVED** [review]
Thanks!

**Pull request merged**